Bugfix/various

MarnuLombard opened the pull request:
* 0f50ec1 [change] Various clean-ups and logic simplifications
* c81ba3b [change] Improve auth error message
* 49da1d9 [change] Delete commented out code
* a0056ed [bugfix] Better error handling and logging
  - Catch all exceptions within out code
  - Never just `die()` and echo the error out
  - Do our best to print a stacktrace for better debugging later
* 7224bd5 [change] Massively simplify logging logic
  - Instead of dancing around whether our `Log` or opencart's has been provided:
    * just make our compatible with theirs
* 415d021 [bugfix] Prevent undefined index errors in case no addresses are returned from API
* 88695b6 [bugfix] Prevent undefined index errors in case no waybill could be booked
  - Create a domain-specific exception and throw it when needed
  - Else an inconsistent return structure is given
* cffc209 [bugfix] Return the correct data structure from `addAddress()`
  - The `contact_id` was previously an array
* 50ea393 [change] Use only short array syntax
  - Opencart (and this plugin) supports no lower than PHP 5.4
* 72ffd57 [bugfix] Downgrade and harden the javascript
  - We can't assume that all browsers support ES6 templates and object destructuring
  - Catch ajax errors
    * Use `alert()` for the time being, no ready-made alert components are available to us
* d75f2b4 [change] Delete a repeated modification in `install.xml`
* ebdbc85 [change] Minor further clean-ups to docblocks
* 63646a6 [change] Bump version number && new zip file
* be2e139 [bugfix] Use `phpversion()` instead of `PHP_VERSION_ID`
  - `PHP_VERSION_ID` was only introduced in PHP 7.4
* d1c8951 [change] Bump minor version number 
  - The fixes are vast enough that they don't just constitute a small `patch` (bugfix) release
  - Rather up the `minor` (backwards incompatible) version
 * 5292b78 New zip file

MarnuLombard committed:
Bump version number

MarnuLombard committed:
[change] Various clean-ups and logic simplifications

MarnuLombard committed:
[change] Improve auth error message

MarnuLombard committed:
[change] Delete commented out code

MarnuLombard committed:
[bugfix] Better error handling and logging
- Catch all exceptions within out code
- Never just `die()` and echo the error out
- Do our best to print a stacktrace for better debugging later
-

MarnuLombard committed:
[change] Massively simplify logging logic
- Instead of dancing around whether our `Log` or opencart's has been provided:
  * just make our compatible with theirs

MarnuLombard committed:
[bugfix] Prevent undefined index errors in case no addresses are returned from API

MarnuLombard committed:
[bugfix] Prevent undefined index errors in case no waybill could be booked
- Create a domain-specific exception and throw it when needed
- Else an inconsistent return structure is given

MarnuLombard committed:
[bugfix] Return the correct data structure from `addAddress()`
- The `contact_id` was previously an array

MarnuLombard committed:
[change] Use only short array syntax
- Opencart (and this plugin) supports no lower than PHP 5.4

MarnuLombard committed:
[bugfix] Downgrade and harden the javascript
- We can't assume that all browsers support ES6 templates and object destructuring
- Catch ajax errors
  * Use `alert()` for the time being, no ready-made alert components are available to us

MarnuLombard committed:
[change] Delete a repeated modification in `install.xml`

MarnuLombard committed:
[change] Minor further clean-ups to docblocks

MarnuLombard committed:
[change] Bump version number && new zip file

WilRyan commented on install.xml:
```diff
@@ -258,13 +258,19 @@
                             $this->collivery->acceptCollivery($waybill_id);
                         }
+                    } catch (\Mds\ColliveryException $e) {
+                        $this->response->addHeader($this->request->server['SERVER_PROTOCOL'] . '/1.1 400 Bad Request');
```
can you assume all are 400 errors?

MarnuLombard replied:
That's very good point, no we can't it could very well be an error in the Collivery API.
*however*, given the crude frontend implementation that is consuming this (`$.ajax()`) -
It doesn't make a difference what the error code is - as long as the status is an error  and `jQuery` enters the `fail()` method of the ajax call.

MarnuLombard committed:
[bugfix] Use `phpversion()` instead of `PHP_VERSION_ID`
- `PHP_VERSION_ID` was only introduced in PHP 7.4

MarnuLombard committed:
[change] Bump minor version number
- The fixes are vast enough that they don't just constitute a small `patch` (bugfix) release
- Rather up the `minor` (backwards incompatible) version

MarnuLombard committed:
[change] New zip file
